Lib/test/test_operator.py

```diff
@@ -533,6 +573,13 @@ def __length_hint__(self):
         with self.assertRaises(LookupError):
             operator.length_hint(X(LookupError))
 
+        class Y: pass
```

Comment:
Here the Python and C versions will behave differently if `__length_hint__` returns an instance of a subclass of int.

Python:
```
>>> class X:
...     def __length_hint__(self): return True
... 
>>> operator.length_hint(X())
True
```

C:
```
>>> class X:
...     def __length_hint__(self): return True
... 
>>> operator.length_hint(X())
1
```

Arguably a bug but again I don't know if it's worth complicating the code for.